### PR TITLE
Move non-core DNS zones outside network module

### PR DESCRIPTION
**Pull request by tamirkamara**
## What is being addressed

We create many objects in the network module that cause delays on its completion but aren't really required to be provisioned there.

## How is this addressed

- Split out all DNS zones not required for the core applications, and move them out of the network module. This allows TF to better manage dependencies and on some scenarios expedite deployment (as many things are waiting for the network module).

### CODE REVIEW

File: core/terraform/locals.tf [hunk 1]
```diff
@@ -12,4 +12,24 @@ locals {
...
+  private_dns_zone_names_non_core = toset([
+    "privatelink.purview.azure.com",
+    "privatelink.purviewstudio.azure.com",
+    "privatelink.sql.azuresynapse.net",
+    "privatelink.dev.azuresynapse.net",
+    "privatelink.azuresynapse.net",
+    "privatelink.dfs.core.windows.net",
+    "privatelink.azurehealthcareapis.com",
+    "privatelink.dicom.azurehealthcareapis.com",
+
```
Comment (guybartal) [inline core/terraform/locals.tf:27]: is there a reason for this empty line?

Comment (tamirkamara) [reply core/terraform/locals.tf:27]: there was but I don't remember. I'll remove it.

File: core/terraform/migrate.sh [hunk 2]
```diff
@@ -189,6 +189,43 @@ if [ -n "${rp_subnet}" ]; then
...
+do
+  dns_zone_rid=$(echo "${terraform_show_json}" \
+    | jq -r "select(.values.root_module.child_modules != null) .values.root_module.child_modules[] | select (.address==\"module.network\") | .resources[] | select(.address==\"${zones[$value]}\") | .values.id")
+  if [ -n "${dns_zone_rid}" ]; then
+    terraform state rm "${zones[$value]}"
+    terraform import azurerm_private_dns_zone.non_core[\""$value"\"] "${dns_zone_rid}"
+  fi
+done
+
+# module.network.azurerm_private_dns_zone_virtual_network_link.mysql
```
Comment (guybartal) [inline core/terraform/migrate.sh:211]: maybe change comment to `import state if module.network.azurerm_private_dns_zone_virtual_network_link.mysql exists`

**Review by guybartal — APPROVED** [review]
LGTM

File: core/terraform/migrate.sh (hunk 2 above)
Comment (tamirkamara) [reply core/terraform/migrate.sh:211]: I'll delete it too since every single block works in the same way here.

**Pull request merged**